### PR TITLE
Backport of MAGETWO-60351 for Magento 2.1 - Unnecessary disabled paym…

### PR DESCRIPTION
**Pull request by hostep**
…ent methods on checkout page #4868

### Description
This is a backport of issue MAGETWO-60351 for Magento 2.1

### Fixed Issues (if relevant)
1. Partly https://github.com/magento/magento2/issues/4868: Checkout page very large and quite slow.

Reference to commits on the develop branch: https://github.com/magento/magento2/commit/398b91532a3abdfd29c9d02749e41ac40b20a611 & https://github.com/magento/magento2/commit/79f392c4f1cf43675acc09fcc80ac248a40e1635 & https://github.com/magento/magento2/commit/0644f24c7d658a23733c51f670c1f6931be39a6f

### Discussion
Together with the [backport of MAGETWO-59685](https://github.com/magento/magento2/pull/9364), this significantly speeds up the checkout process, even up to multiple seconds. This is due to the json data structure being outputted on the checkout containing way too many elements then necessary, which makes the client side processing of this json take a very long time (sometimes up to 10 seconds). In a concrete example of one of our webshops, this reduced the json structure from 175867 lines to 4003 lines.

MAGETWO-60351 in particular fixes not outputting payment methods which your webshop isn't using.

### CODE REVIEW

File: app/code/Magento/Payment/Plugin/PaymentConfigurationProcess.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
```
Comment (omiroshnichenko) [inline app/code/Magento/Payment/Plugin/PaymentConfigurationProcess.php]: Update copyright.

File: app/code/Magento/Payment/Test/Unit/Plugin/PaymentConfigurationProcessTest.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
```
Comment (omiroshnichenko) [inline app/code/Magento/Payment/Test/Unit/Plugin/PaymentConfigurationProcessTest.php]: Same problem with copyright.

File: app/code/Magento/Vault/Plugin/PaymentVaultConfigurationProcess.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
```
Comment (omiroshnichenko) [inline app/code/Magento/Vault/Plugin/PaymentVaultConfigurationProcess.php]: Same problem with copyright.

File: app/code/Magento/Vault/Test/Unit/Plugin/PaymentVaultConfigurationProcessTest.php [hunk 4]
```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Copyright © 2016 Magento. All rights reserved.
```
Comment (omiroshnichenko) [inline app/code/Magento/Vault/Test/Unit/Plugin/PaymentVaultConfigurationProcessTest.php]: Same problem with copyright.

**Comment by hostep** [comment]
@omiroshnichenko: I just amended the commit with the changed copyright header, and force pushed it.

Thanks for the heads up!

**Review by ishakhsuvarov — APPROVED** [review]

**Comment by magento-team** [comment]
@hostep thank you for your contribution to Magento 2 project

**Pull request merged**